clientv3/retry.go

```diff
@@ -36,6 +36,14 @@ func (c *Client) newRetryWrapper() retryRpcFunc {
 			eErr := rpctypes.Error(err)
 			// always stop retry on etcd errors
 			if _, ok := eErr.(rpctypes.EtcdError); ok {
+				if grpc.Code(err) != codes.Unavailable {
+					// try to request another endpoint at next rpc
+					// leave this not retry
+					// already do anti-too-frequently switch
+					// at most switch one time per 30 seconds
+					// can be securely called by concurrently
+					c.SwitchEndpoint()
+				}
```

Comment:
* about the put/txn 's only-once semantic, only do switch here but not retry. keep this request fail and wish next time will success